rest/model/admin.go

```diff
@@ -118,20 +118,38 @@ func (as *APIAdminSettings) BuildFromService(h interface{}) error {
 // ToService returns a service model from an API model
 func (as *APIAdminSettings) ToService() (interface{}, error) {
 	settings := evergreen.Settings{
-		ApiUrl:             *as.ApiUrl,
-		Banner:             *as.Banner,
-		BannerTheme:        evergreen.BannerTheme(*as.BannerTheme),
-		ClientBinariesDir:  *as.ClientBinariesDir,
-		ConfigDir:          *as.ConfigDir,
-		Credentials:        map[string]string{},
-		Expansions:         map[string]string{},
-		GithubPRCreatorOrg: *as.GithubPRCreatorOrg,
-		IsNonProd:          *as.IsNonProd,
-		Keys:               map[string]string{},
-		LogPath:            *as.LogPath,
-		Plugins:            evergreen.PluginConfig{},
-		PprofPort:          *as.PprofPort,
-		SuperUsers:         as.SuperUsers,
+		Credentials: map[string]string{},
+		Expansions:  map[string]string{},
+		Keys:        map[string]string{},
+		Plugins:     evergreen.PluginConfig{},
+		SuperUsers:  as.SuperUsers,
+	}
+	if as.ApiUrl != nil {
```

Comment:
Side note, I keep having to do this a lot. Should we just be using recover() for this sort of thing instead?

Reply:
Personally I don't use an error handler unless I can't anticipate the types of errors that may be encountered. I think this is more robust

Reply:
Fair point. I find it super tedious when interacting with REST apis.